tokio/src/net/windows/named_pipe.rs

```diff
@@ -2588,13 +2619,13 @@ mod test {
 
         opts.reject_remote_clients(false);
         opts.pipe_mode(PipeMode::Message);
-        assert_eq!(opts.pipe_mode, PIPE_TYPE_MESSAGE);
+        assert_eq!(opts.pipe_mode, PIPE_TYPE_MESSAGE | PIPE_READMODE_MESSAGE);
 
         opts.reject_remote_clients(true);
         opts.pipe_mode(PipeMode::Message);
         assert_eq!(
             opts.pipe_mode,
-            PIPE_TYPE_MESSAGE | PIPE_REJECT_REMOTE_CLIENTS
+            PIPE_TYPE_MESSAGE | PIPE_READMODE_MESSAGE | PIPE_REJECT_REMOTE_CLIENTS
         );
```

Comment:
What happens here if we change the pipe mode back to byte? Is `PIPE_READMODE_MESSAGE` still set?

Reply:
`PIPE_READMODE_MESSAGE` is cleared. I've pushed c582f04c to add that to the test. :)

I have another commit (658c3f85f710bb039a4b097cf2ed8808def92ddc) queued after this to refactor the code to not use bitfields (#5359).